api/src/main/java/org/openmrs/api/db/ContextDAO.java

```diff
@@ -62,6 +63,32 @@
 	 */
 	public User getUserByUuid(String uuid) throws ContextAuthenticationException;
 	
+	/**
+	 * Gets a user given the username. Privilege checks are not done here because this is used by the
+	 * {@link Context#getAuthenticatedUser()} or {@link Context#authenticate(org.openmrs.api.context.Credentials)} methods.
+	 * 
+	 * @param username The username of the user to fetch
+	 * @return The matched user, null otherwise.
+	 * 
+	 * @since 2.3.0
+	 */
+	public User getUserByUsername(String username);
```

Comment:
Where is this method used?

Reply:
@dkayiwa auth. schemes will need the ability to fetch users (typically to see if they already exist or not) to populate `Authenticated` implementations that then are able to return the authenticated user (see [here](https://github.com/openmrs/openmrs-core/blob/28990338f2096229268ac668d367bedce96d3d63/api/src/main/java/org/openmrs/api/context/Authenticated.java#L30)). In all likelihood users will be known in third-party auth. providers through their SSO usernames, this is why this new API is needed.